opnode/l2/api.go

```diff
@@ -1,3 +1,4 @@
+// Package l2 connects to the L2 execution engine over the Catalyst API.
```

Comment:
```suggestion
// Package l2 connects to the L2 execution engine over the Engine API.
```
"catalyst" is a legacy name used inside of geth for the Merge related parts (the first Geth merge prototype by Guillaume had this name), "Engine API" is better